balancer-js/src/modules/data/pool/subgraph.ts

```diff
@@ -23,7 +23,7 @@ export class SubgraphPoolRepository implements PoolRepository {
         const { pool0 } = await this.client.Pools({
           where: { address: value },
         });
-        return this.mapPool(pool0);
+        return this.mapPool(pool0[0]);
```

Comment:
Is this intentional?

Reply:
Yes, this was already fixed on another branch. `pool0` returns an array and this find method should return just the first item if it exists. `undefined` is handled in the `mapPool` method, so no need to throw errors as well.

Reply:
This is actually wrong. I was considering the findBy `address` to behave the same way as findBy `id`, and this is not the case. findBy `id` returns either `Pool` or `undefined`, and findBy `address` returns either `Pool[]` or `[]`. I'll refactor it. Btw, migrations branch has the same issue and I'll fix it there as well.

Reply:
In fact, I just saw that this was completely reimplemented on the APRs branch. The effective result for the scenario we're currently working on is the same, so I'm not sure if it's worth updating it on this branch.